### PR TITLE
DEV-31040 fix

### PR DESCRIPTION
**Pull request by pcdeshmukh**
Using existing selectRanges to get editor ranges and using underlying api to reflect replaced changes

Also adding changes to run tests locally.

### CODE REVIEW

File: src/adapters/CKEditor5Adapter.ts [hunk 1]
```diff
@@ -73,14 +80,36 @@ export class CKEditor5Adapter extends AbstractRichtextEditorAdapter {
...
+
+    await new Promise<void>((resolve) => {
+      editor.model.document.selection.once('change', (..._args) => {
+        resolve();
+      });
+    });
+
+    editor.model.change((writer) => {
+      const selectedRange = editor.model.document.selection.getFirstRange();
+      if (selectedRange) {
```
Comment (abhijeetnarvekar) [inline src/adapters/CKEditor5Adapter.ts:107]: Maybe we should check for selected range end > start to be sure selection is present.
The odds could be some other event got in between our calls which can possibly remove the selection we made

**Review by abhijeetnarvekar — CHANGES_REQUESTED** [review]

File: src/adapters/CKEditor5Adapter.ts (hunk 1 above)
Comment (pcdeshmukh) [reply src/adapters/CKEditor5Adapter.ts:107]: what about inserting text at certain location right now we are not getting such matches but it's possible. In that case end == start.

Comment (pcdeshmukh) [reply src/adapters/CKEditor5Adapter.ts:107]: As discussed for insertion we should have other api then replace.

**Review by abhijeetnarvekar — DISMISSED** [review]

**Comment by abhijeetnarvekar** [comment]
Obsolete now

**Pull request closed without merging**